core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -292,26 +293,28 @@ protected Node visitShowRoles(ShowRoles node, Void context)
         {
             Optional<String> catalog = node.getCatalog().map(c -> c.getValue().toLowerCase(ENGLISH));
 
+            List<Expression> rows;
             if (node.isCurrent()) {
                 accessControl.checkCanShowCurrentRoles(session.toSecurityContext(), catalog);
                 Set<String> enabledRoles = catalog.map(c -> metadata.listEnabledRoles(session, c))
                         .orElseGet(() -> session.getIdentity().getEnabledRoles());
-                List<Expression> rows = enabledRoles.stream()
+                rows = enabledRoles.stream()
                         .map(role -> row(new StringLiteral(role)))
                         .collect(toList());
-                return simpleQuery(
-                        selectList(new AllColumns()),
-                        aliased(new Values(rows), "role_name", ImmutableList.of("Role")));
             }
             else {
                 accessControl.checkCanShowRoles(session.toSecurityContext(), catalog);
-                List<Expression> rows = metadata.listRoles(session, catalog).stream()
+                rows = metadata.listRoles(session, catalog).stream()
                         .map(role -> row(new StringLiteral(role)))
                         .collect(toList());
-                return simpleQuery(
-                        selectList(new AllColumns()),
-                        aliased(new Values(rows), "role_name", ImmutableList.of("Role")));
             }
+            List<String> columns = ImmutableList.of("Role");
+            if (rows.isEmpty()) {
+                return emptyQuery(columns);
```

Comment:
Do I understand correctly that this was impossible situation for the Connector Roles (when System Access roles were not implemented) to return the empty result?

Reply:
No, it was implemented differently. It was reading previously from a `information_schema.roles`, now we are using `VALUES`